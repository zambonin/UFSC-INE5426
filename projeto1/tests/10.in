int a(10)
int i = 0
int ref mypointer
int ref pointers(2)
mypointer = addr i
i = ref mypointer + 1
pointers(0) = mypointer
pointers(1) = addr a(3)
int ref ref doublepointer
doublepointer = addr pointers(0)

int i, j = 2
int ref ip, jp
ip = addr i
jp = addr j
ref ip = ref jp

