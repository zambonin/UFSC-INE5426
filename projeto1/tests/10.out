int array: a (size: 10)
int var: i = 0
int ref var: mypointer
int ref array: pointers (size: 2)
= mypointer [addr] i
= i + [ref] mypointer 1
= [index] pointers 0 mypointer
= [index] pointers 1 [addr] [index] a 3
int ref ref var: doublepointer
= doublepointer [addr] [index] pointers 0
int var: i, j = 2
int ref var: ip, jp
= ip [addr] i
= jp [addr] j
= [ref] ip [ref] jp

